hazelcast-coverage-report/dependencies-list.sh

```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+mvn -f .. -T 1 -B exec:exec -Dexec.executable=echo -Dexec.args='#Module-info:<dependency><groupId>${project.groupId}</groupId><artifactId>${project.artifactId}</artifactId><version>$${project.version}</version><type>pom</type></dependency>' | grep '#Module-info:' | grep -v 'hazelcast-coverage-report' | cut -f2 -d':'
```

Comment:
Good point. 
In practice, we need the same modules as during sonar analysis. 
It means that my script can lie if used JDK version is different than for the sonar analysis. On the other hand, we cannot add any dependency which won't be available during regular or sonar builds like, for example, the "jdk17-tests" module as it will fail then.
I don't think it makes sense to create some profiles etc., to cover the above.
Maybe it would be better to add proper information in the:
* Jenkins jobs - to remember about changing the JDK version in both places - modules validator + in the job which runs sonar analysis (currently, we are using JDK11)
* Change comment in pom.xml of hazelcast-coverage-module that we need all modules which are available will be run for sonar analysis - not all that are exist
* Add a comment in my script that output can differ for different JDK versions.

WDYT?

Reply:
:+1: The comments would make it clear.